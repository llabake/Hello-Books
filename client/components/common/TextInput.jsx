import React from 'react';
import PropTypes from 'prop-types';


const TextInput = ({ id, type, icon, placeholder, name, errors, value, onChange, onBlur }) => {
  return (
    <div className='input-field col s12'>
      <i className="material-icons prefix">{icon}</i>
      <input 
      id={id} 
      type={type} 
      required
      placeholder={placeholder}
      className="validate"
      name={name}
      value={value} 
      onChange={onChange}
      onBlur={onBlur}
      />
      {errors && errors.length ?
        errors.map((error, i) => { return (
        <div key={i} 
        className='red-text left-align' 
        style={{ 
          'fontSize':'medium', 
          'marginLeft': 46, 
          'marginTop': -12
          }}>
          {error}
        </div>
        )}) : null }
    </div>
  )
}

TextInput.propTypes = {
  type: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  errors: PropTypes.array,
  placeholder: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func.isRequired,
};

export default TextInput;

