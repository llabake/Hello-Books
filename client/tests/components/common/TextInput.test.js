import React from 'react';

import TextInput from '../../../components/common/TextInput';

test('Text input field component renders correctly', () => {
  const props = {
    id: '',
    type: '',
    icon: '',
    placeholder: '',
    name: '',
    errors: [],
    value: '',
    onChange: jest.fn(),
    onBlur: jest.fn()
  }
  const wrapper = mount(<TextInput {...props} />);
  expect(wrapper).toBeDefined();
  expect(wrapper).toMatchSnapshot();
})

test('Text input field component show error when they occur', () => {
  const props = {
    id: '',
    type: '',
    icon: '',
    placeholder: '',
    name: '',
    errors: [
      'Username is required'
    ],
    value: '',
    onChange: jest.fn(),
    onBlur: jest.fn()
  }
  const wrapper = mount(<TextInput {...props} />);
  expect(wrapper).toBeDefined();
  expect(wrapper.find('div.red-text').exists()).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
})