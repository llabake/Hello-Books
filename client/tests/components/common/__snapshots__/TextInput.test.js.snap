// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text input field component renders correctly 1`] = `
<TextInput
  errors={Array []}
  icon=""
  id=""
  name=""
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  placeholder=""
  type=""
  value=""
>
  <div
    className="input-field col s12"
  >
    <i
      className="material-icons prefix"
    />
    <input
      className="validate"
      id=""
      name=""
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      placeholder=""
      required={true}
      type=""
      value=""
    />
  </div>
</TextInput>
`;

exports[`Text input field component show error when they occur 1`] = `
<TextInput
  errors={
    Array [
      "Username is required",
    ]
  }
  icon=""
  id=""
  name=""
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  placeholder=""
  type=""
  value=""
>
  <div
    className="input-field col s12"
  >
    <i
      className="material-icons prefix"
    />
    <input
      className="validate"
      id=""
      name=""
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      placeholder=""
      required={true}
      type=""
      value=""
    />
    <div
      className="red-text left-align"
      key="0"
      style={
        Object {
          "fontSize": "medium",
          "marginLeft": 46,
          "marginTop": -12,
        }
      }
    >
      Username is required
    </div>
  </div>
</TextInput>
`;
